### ISSUE TITLE
warning: option --root is deprecated; use --sysroot instead

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/manjaro-tools#294]
Currently we fail on pacman 5.1 series with:

```
==> Creating clean working copy [phil]...done
chmod: changing permissions of '/tmp/tmp.t4ieEXfwRY': Operation not permitted

==> ERROR: An unknown error has occurred. Exiting...

==> ERROR: An unknown error has occurred. Exiting...
/usr/bin/mkchrootpkg: line 255: 14370 User defined signal 1   sudo -u $SUDO_USER env SRCDEST="$SRCDEST" BUILDDIR="$builddir" makepkg --config="$copydir/etc/makepkg.conf" --verifysource -o
```

**Comment by oberon2007** [comment]
chowning the tmp dir at least lets me build pkgs.
**/usr/bin/mkchroot**, line 255
```
download_sources() {
    local builddir="$(mktemp -d)"
    chown $SUDO_USER "$builddir"
    chmod 1777 "$builddir"
```

**Comment by oberon2007** [comment]
where are we currently using `--root`?

**Comment by oberon2007** [comment]
Right. We are talking about `pacman -r` which is currently called by [basestrap](f87fa4be1e051126530f33d5e568ee50075026fd)
Just replacing option `-r` with `--sysroot` isn't enough because that leads **buildiso** fail with
```
  -> Installing packages to /var/lib/manjaro-tools/buildiso/i3/x86_64/rootfs
error: config file /var/lib/manjaro-tools/buildiso/i3/x86_64/rootfs/pacman-basestrap.conf could not be read: No such file or directory
```
So. Needs further investigation.
However, the deprecation warning doesn't seem to cause a failure right now.

**Comment by oberon2007** [comment]
Some information can be found [here](http://allanmcrae.com/2018/05/pacman-5-1-dont-use-the-force-luke/),
namely:
> We have also added a --sysroot option that will replace --root. Basically, this now works the way people will expect – for example, the configuration file used is the one in the specified root, and not the local one. This does require a bit more setup while creating a new install root, but hopefully will be a lot more robust.

We just still need to establish what kind of preparation will be needed in `$newroot` ...

**Comment by oberon2007** [comment]
Stange thing is, that as stated above, when you replace `pacman -r` with `pacman --sysroot` in **/usr/bin/basestrap**, buildiso claims 
```
config file /var/lib/manjaro-tools/buildiso/i3/x86_64/rootfs/pacman-basestrap.conf could not be read: No such file or directory
```
while in fact **/var/lib/manjaro-tools/buildiso/i3/x86_64/rootfs/pacman-basestrap.conf** is actually present!

**Comment by philmmanjaro** [comment]
Similar issue here: https://bugs.archlinux.org/task/58778
